lib/pbench/server/indexing_tarballs.py

```diff
@@ -0,0 +1,416 @@
+
+        self.options = options
+        _re_idx = "RE-" if options.re_index else ""
+        if options.index_tool_data:
+            # The link source and destination for the operation of this script
+            # when it only indexes tool data.
+            self.linksrc = "TO-INDEX-TOOL"
+            self.linkdest = "INDEXED"
+        else:
+            # The link source and destination for the operation of this script
+            # when it indexes run, table-of-contents, and result data.
+            self.linksrc = f"TO-{_re_idx}INDEX"
+            self.linkdest = "TO-INDEX-TOOL"
+        # We only ever use a symlink'd error destination for indexing
+        # problems.
+        self.linkerrdest = "WONT-INDEX"
+        self.idxctx = idxctx
+        self.incoming = incoming
+        self.name = name
+
+    def collect_tb(self, ARCHIVE_rp, qdir):
+        """ Collect tarballs that needs indexing"""
+
+        # find -L $ARCHIVE/*/$linksrc -name '*.tar.xz' -printf "%s\t%p\n" 2>/dev/null | sort -n > $list
+        tarballs = []
+        idxctx = self.idxctx
+        try:
+            tb_glob = os.path.join(ARCHIVE_rp, "*", self.linksrc, "*.tar.xz")
+            for tb in glob.iglob(tb_glob):
+                try:
+                    rp = Path(tb).resolve(strict=True)
+                except OSError:
+                    idxctx.logger.warning("{} does not resolve to a real path", tb)
+                    quarantine(qdir, idxctx.logger, tb)
+                    continue
+                controller_path = rp.parent
+                controller = controller_path.name
+                archive_path = controller_path.parent
+                if str(archive_path) != str(ARCHIVE_rp):
+                    idxctx.logger.warning(
+                        "For tar ball {}, original home is not {}", tb, ARCHIVE_rp
+                    )
+                    quarantine(qdir, idxctx.logger, tb)
+                    continue
+                if not Path(f"{rp}.md5").is_file():
+                    idxctx.logger.warning("Missing .md5 file for {}", tb)
+                    quarantine(qdir, idxctx.logger, tb)
+                    # Audit should pick up missing .md5 file in ARCHIVE directory.
+                    continue
+                try:
+                    # get size
+                    size = rp.stat().st_size
+                except OSError:
+                    idxctx.logger.warning("Could not fetch tar ball size for {}", tb)
+                    quarantine(qdir, idxctx.logger, tb)
+                    # Audit should pick up missing .md5 file in ARCHIVE directory.
+                    continue
+                else:
+                    tarballs.append((size, controller, tb))
+        except Exception:
+            idxctx.logger.exception(
+                "Unexpected error encountered generating list"
+                " of tar balls to process"
+            )
+            # tuple to return the status and return value
+            return (1, 12)
```

Comment:
Change this to `return (1, [])`, the second tuple always being a list, empty or otherwise.

Reply:
ok